docs/literate/src/files/subcell_shock_capturing.jl

```diff
@@ -0,0 +1,285 @@
+# variables you want to limit. So, to limit the density with lower and upper local bounds pass
+# the following.
+local_minmax_variables_cons = ["rho"]
+
+# ## Exemplary simulation
+# How to set up a simulation using the IDP limiting becomes clearer when lokking at a exemplary
+# setup. This will be a simplified version of `tree_2d_dgsem/elixir_euler_blast_wave_sc_subcell.jl`.
+# Since the setup is mostly very similar to a pure DGSEM setup as in
+# `tree_2d_dgsem/elixir_euler_blast_wave.jl`, the equivalent parts are without any explanation
+# here.
+using OrdinaryDiffEq
+using Trixi
+
+equations = CompressibleEulerEquations2D(1.4)
+
+function initial_condition_blast_wave(x, t, equations::CompressibleEulerEquations2D)
+    ## Modified From Hennemann & Gassner JCP paper 2020 (Sec. 6.3) -> "medium blast wave"
+    ## Set up polar coordinates
+    inicenter = SVector(0.0, 0.0)
+    x_norm = x[1] - inicenter[1]
+    y_norm = x[2] - inicenter[2]
+    r = sqrt(x_norm^2 + y_norm^2)
+    phi = atan(y_norm, x_norm)
+    sin_phi, cos_phi = sincos(phi)
+
+    ## Calculate primitive variables
+    rho = r > 0.5 ? 1.0 : 1.1691
+    v1 = r > 0.5 ? 0.0 : 0.1882 * cos_phi
+    v2 = r > 0.5 ? 0.0 : 0.1882 * sin_phi
+    p = r > 0.5 ? 1.0E-3 : 1.245
+
+    return prim2cons(SVector(rho, v1, v2, p), equations)
+end
+initial_condition = initial_condition_blast_wave;
+
+# Since the surface integral is equal for both the DG and the subcell FV method, the limiting is
+# applied only in the volume integral.
+#-
+# Note, that the DG method is based on the flux differencing formulation. Hence, you have to use a
+# two-point flux, such as [`flux_ranocha`](@ref), [`flux_shima_etal`](@ref), [`flux_chandrashekar`](@ref)
+# or [`flux_kennedy_gruber`](@ref), for the DG volume flux.
+surface_flux = flux_lax_friedrichs
+volume_flux = flux_ranocha
+
+# The limiter is implemented within [`SubcellLimiterIDP`](@ref). It always requires the
+# parameters `equations` and `basis`. With additional parameters (described [above](@ref IDPLimiter)
+# or listed in the docstring) you can specify and enable additional limiting options.
+# Here, the simulation should contain local limiting for the density using lower and upper bounds.
+basis = LobattoLegendreBasis(3)
+limiter_idp = SubcellLimiterIDP(equations, basis;
+                                local_minmax_variables_cons = ["rho"])
+
+# The initialized limiter is passed to `VolumeIntegralSubcellLimiting` in addition to the volume
+# fluxes of the low-order and high-order scheme.
+volume_integral = VolumeIntegralSubcellLimiting(limiter_idp;
+                                                volume_flux_dg = volume_flux,
+                                                volume_flux_fv = surface_flux)
+
+# Then, the volume integral is passed to `solver` as it is done for the standard flux-differencing
+# DG scheme or the element-wise limiting.
+solver = DGSEM(basis, surface_flux, volume_integral)
+
+#-
```

Comment:
```suggestion
```

Reply:
I would like to keep this to have the solver output in the tutorial
![Screenshot from 2024-05-07 09-59-00](https://github.com/trixi-framework/Trixi.jl/assets/74359358/43dd6bdf-3c2c-42c4-a718-dfc39bb7298a)

Reply:
Ok, but is for this the `#-` in line 166 required?

Reply:
Yes. It is basically a line break.